File: lc-dc-sep/lc-dc-sep14/solution.py
# Problem: 2419. Longest Subarray With Maximum Bitwise AND

class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        target = max(nums)
        res, cnt = 0, 0
        for n in nums:
            if n == target:
                cnt += 1
            else:
                cnt = 0
            res = max(res, cnt)
        
        return res

File: lc-dc-sep/lc-dc-sep14/solution.cpp
// Problem: 2419. Longest Subarray With Maximum Bitwise AND

class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int target = 0;
        for(auto& n: nums){
            if(n>target) target = n;
        }
        int res = 0, cnt = 0;
        for(auto& n: nums){
            if(n == target) cnt++;
            else cnt = 0;
            res = max(res, cnt);
        }
        return res;
    }
};